src/cpu/x64/gemm_bf16_inner_product.hpp

```diff
@@ -74,11 +74,11 @@ struct gemm_bf16_inner_product_fwd_t : public primitive_t {
                     attr()->post_ops_.check_sum_consistency(
                             dst_md()->data_type, /* is_int8 */ false),
                     VERBOSE_UNSUPPORTED_POSTOP);
+            VDISPATCH_INNER_PRODUCT(set_default_params() == status::success,
+                    VERBOSE_UNSUPPORTED_TAG);
```

Comment:
Does this fix an issue when dst is any ? If so, shall we add a test case to catch it if it was not already caught?

(nit) maybe move this up after bias check and before post-ops mask check to keep post-ops related checks grouped?

Reply:
It does, and there's a case to catch it, the thing is it's a dispatching issue. But the implementation itself will be triggered for bf16 on non-bf16 native ISA, so I'm not sure how important it is...

I can group them, sure.